dev/PushNotifications/PushNotifications.idl

```diff
@@ -0,0 +1,134 @@
+    {
+        // Initialize using the IBackgroundInstance: used specifically for the Background Activation scenario
+        static PushNotificationReceivedEventArgs CreateFromBackgroundTaskInstance(Windows.ApplicationModel.Background.IBackgroundTaskInstance backgroundTask);
+
+// Initialize using the PushNotificationEventArgs from Windows: used specifically for in-memory event handling when app is already in foreground
+static PushNotificationReceivedEventArgs CreateFromPushNotificationReceivedEventArgs(Windows.Networking.PushNotifications.PushNotificationReceivedEventArgs args);
+
+// The Push payload
+byte[] Payload { get; };
+
+// Gets a deferral to run under specific modes like low power mode
+Windows.ApplicationModel.Background.BackgroundTaskDeferral GetDeferral();
+
+// Subscribe to Cancelled event handler to be signalled when resource policies are no longer true like 30s wallclock timer
+event Windows.ApplicationModel.Background.BackgroundTaskCanceledEventHandler Canceled;
+
+// Set to true to prevent proceeding launch due to Background Activation: false by default
+Boolean Handled;
+};
+
+[flags]
+enum PushNotificationRegistrationKind
+{
+    PushTrigger = 0x1, // Registers a Push Trigger with Background Infrastructure
+    ComActivator = 0x2, // Registers the Project Reunion Background Task component as an InProc COM server 
+};
+
+// An abstraction over the activation Registration flow
+runtimeclass PushNotificationActivationInfo
+{
+    // Initialize using a RegistrationKind and optionally defined parameters like manifest defined activatorId
+    // 1) If kind = PushTrigger is specified, only the Push Trigger will be Registered with Background Infra
+    // 2) If kind = ComActivator is specified, the Project Reunion Background Task component will be Registered as an InProc COM server
+    PushNotificationActivationInfo(PushNotificationRegistrationKind kind, Guid taskClsid);
+
+// The CLSID associated with the Client COM server that Project Reunion will activate
+Guid TaskClsid{ get; };
+
+PushNotificationRegistrationKind Kind{ get; };
+
+// The conditions under which Push Triggers would execute
+Windows.ApplicationModel.Background.IBackgroundCondition[] GetConditions();
+void SetConditions(Windows.ApplicationModel.Background.IBackgroundCondition[] conditions);
+};
+
+enum PushNotificationChannelStatus
+{
+    InProgress, // The request is in progress and there is no retry operation
+    InProgressRetry, // The request is in progress and is in a backoff retry state. Check ExtendedError for HRESULT for retryable error.
+    CompletedSuccess, // The request completed successfully
+    CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+};
+
+// The PushNotificationChannel Progress result
+struct PushNotificationCreateChannelStatus
+{
+    // The last extended error we failed Channel requests on that caused the inprogress retry status. S_OK if this is just progress status.
```

Comment:
Comment is incorrect. Progress status reports the last error, or `E_PENDING` if operation hasn't yet started - not `S_OK`. But I think we can design away this struct entirely.